module_analysis/post_init_hook.py

```diff
@@ -0,0 +1,16 @@
+# Copyright (C) 2019-Today: GRAP (<http://www.grap.coop/>)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import SUPERUSER_ID, api, tools
+
+
+def analyse_installed_modules(cr, registry):
+    if tools.config["test_enable"]:
+        return
+    with api.Environment.manage():
+        env = api.Environment(cr, SUPERUSER_ID, {})
+        installed_modules = env["ir.module.module"].search(
+            ["|", ("state", "=", "installed"), ("name", "=", "module_analysis")]
+        )
+        installed_modules.button_analyse_code()
```

Comment:
why call this function in each load ?

Reply:
It's called once, right after the module is installed.
It's true that, with the cron you implemented, it may not be needed -- but I guess it's so that users don't have to wait for the cron to run the first time